### PR TITLE
Cheqi add interfaces

### PR DESCRIPTION
**Pull request by MaggieQi**
*add mergeindex from indexblobs
*add search interface to support batch mode
*fix metamapping load issue

### CODE REVIEW

File: AnnService/src/Core/VectorIndex.cpp [hunk 1]
```diff
@@ -114,8 +114,10 @@ VectorIndex::BuildMetaMapping()
 {
     m_pMetaToVec.reset(new std::unordered_map<std::string, SizeType>);
     for (SizeType i = 0; i < m_pMetadata->Count(); i++) {
-        ByteArray meta = m_pMetadata->GetMetadata(i);
-        m_pMetaToVec->emplace(std::string((char*)meta.Data(), meta.Length()), i);
+        if (ContainSample(i)) {
+            ByteArray meta = m_pMetadata->GetMetadata(i);
+            m_pMetaToVec->emplace(std::string((char*)meta.Data(), meta.Length()), i);
```
Comment (SkyInTheSea) [inline AnnService/src/Core/VectorIndex.cpp:119]: if i not contains in m_pMetadata expected? should we add some error handle code?

File: AnnService/src/Core/VectorIndex.cpp [hunk 2]
```diff
@@ -269,9 +276,13 @@ VectorIndex::BuildIndex(std::shared_ptr<VectorSet> p_vectorSet,
 
 
 ErrorCode
-VectorIndex::SearchIndex(const void* p_vector, int p_neighborCount, bool p_withMeta, BasicResult* p_results) const {
-    QueryResult res(p_vector, p_neighborCount, p_withMeta, p_results);
-    SearchIndex(res);
+VectorIndex::SearchIndex(const void* p_vector, int p_vectorCount, int p_neighborCount, bool p_withMeta, BasicResult* p_results) const {
+    size_t vectorSize = GetValueTypeSize(GetVectorValueType()) * GetFeatureDim();
+#pragma omp parallel for schedule(dynamic,10)
+    for (int i = 0; i < p_vectorCount; i++) {
```
Comment (SkyInTheSea) [inline AnnService/src/Core/VectorIndex.cpp:282]: searchindex need follow the policy to leverage only one thread, even for batch search(though fresh ANN index didn't support batch search), as we use request thread pool to control processor usage, every request thread consume several core is not our expectation

File: AnnService/src/Core/VectorIndex.cpp (hunk 1 above)
Comment (MaggieQi) [reply AnnService/src/Core/VectorIndex.cpp:119]: Here we check whether this vector has been deleted or not. If we do not check this, the metadata will be overwritten by deleted items.

File: AnnService/src/Core/VectorIndex.cpp (hunk 2 above)
Comment (MaggieQi) [reply AnnService/src/Core/VectorIndex.cpp:282]: This interface is only used for parallel search in the python and java wrappers to speedup the offline search.

**Pull request merged**